src/test/test-id128.c

```diff
@@ -38,7 +38,7 @@ TEST(id128) {
         assert_se(!sd_id128_in_set(id, ID128_WALDI));
         assert_se(!sd_id128_in_set(id, ID128_WALDI, ID128_WALDI));
 
-        if (sd_booted() > 0) {
+        if (sd_booted() > 0 && access("/etc/machine-id", F_OK) >= 0) {
```

Comment:
Hmm, but wouldn't systemd always create `/etc/machine-id`?

Reply:
it might have been mounted in the chroot from outside, or weird things like that

Reply:
I'm asking in the other direction. I.e. I expect that if the first clause succeeds, the second one will always succeed too.

Reply:
yeah that's what I mean, when building just because there's /run/systemd it doesn't mean much, it might be coming from the host or it might be a leftover

Reply:
I don't think that's possible. `/run` is a tmpfs, so "left over" would mean that systemd was running and is not, which isn't realistic. And if it's mounted in a chroot from outside, that would mean that something is very broken. Something would need to explicitly mount `/run` into the chroot, and mounting host's `/run` into a chroot is a strange mistake to make.

I guess this doesn't matter much, but I think the second clause of the condition is unnecessary.

Reply:
it's happening in the GH CI, although it's not booted